dev-client/src/hooks/useMapSuggestions.ts

```diff
@@ -0,0 +1,95 @@
+import {useCallback, useState} from 'react';
+
+import {useDebouncedCallback} from 'use-debounce';
+
+import {Coords} from 'terraso-client-shared/types';
+
+import {MAP_QUERY_MIN_LENGTH} from 'terraso-mobile-client/constants';
+import {
+  initMapSearch,
+  Suggestion,
+} from 'terraso-mobile-client/services/mapSearchService';
+import {isValidCoordinates} from 'terraso-mobile-client/util';
+
+/*
+ * Logic extracted from screens/HomeScreen/components/MapSearch.tsx; see its revisions
```

Comment:
Oo I haven't been doing this when extracting things into new files but I like it